addons/mrp/models/mrp_production.py

```diff
@@ -576,19 +576,22 @@ def _compute_workorder_ids(self):
 
     @api.depends('state', 'move_raw_ids.state')
     def _compute_reservation_state(self):
-        self.reservation_state = False
+        processed_production = self.env['mrp.production']
         for production in self:
             if production.state in ('draft', 'done', 'cancel'):
                 continue
             relevant_move_state = production.move_raw_ids._get_relevant_state_among_moves()
             # Compute reservation state according to its component's moves.
             if relevant_move_state == 'partially_available':
+                processed_production += production
                 if production.workorder_ids.operation_id and production.bom_id.ready_to_produce == 'asap':
                     production.reservation_state = production._get_ready_to_produce_state()
                 else:
                     production.reservation_state = 'confirmed'
             elif relevant_move_state != 'draft':
+                processed_production += production
                 production.reservation_state = relevant_move_state
+        (self - processed_production).reservation_state = False
```

Comment:
Explain in the commit message why it's needed. Why the compute happens later now compare to before